ui/contexts/snaps/snap-interface.tsx

```diff
@@ -110,9 +110,8 @@ export const SnapInterfaceContextProvider: FunctionComponent<
         params: {
           event: {
             type: event,
-            // TODO: Allow null in the types and simplify this
-            ...(name !== undefined && name !== null ? { name } : {}),
-            ...(value !== undefined && value !== null ? { value } : {}),
+            ...(name === undefined ? {} : { name }),
+            ...(value === undefined ? {} : { value }),
```

Comment:
Are these guaranteed to never be `null` now?

Reply:
I assumed that since these are json-rpc-request params `null` would be a valid type, but if not I can restore the previous code. 

Reverted here: e0f9925915cef9e62512ea5d892951b4fa685662